engines/tika/src/main/java/org/alfresco/transform/tika/parsers/ExifToolParser.java

```diff
@@ -76,13 +77,34 @@ public class ExifToolParser extends ExternalParser {
     private String separator;
 
     public ExifToolParser() {
+        this(null);
+    }
+
+    public ExifToolParser(RuntimeExec exifRuntimeExec) {
         super();
         try {
             List<ExternalParser> eParsers = ExternalParsersFactory.create(getExternalParserConfigURL());
             // if ExifTool is not installed then no parsers are returned
             if (eParsers.size() > 0) {
                 ExternalParser eParser = eParsers.get(0);
-                this.setCommand(eParser.getCommand());
+
+                String[] commandToBeExecuted;
+                if( exifRuntimeExec==null ) {
+                    logger.debug("Command to be executed determined from Tika ExternalParser");
+                    commandToBeExecuted = eParser.getCommand();
+                } else {
+                    logger.debug("Command to be executed determined from RuntimeExec");
+                    commandToBeExecuted = exifRuntimeExec.getCommand();
+                }
+                // give some last chance

```

Comment:
Please replace this comment with a blank line. It add nothing to the understanding and I was left wondering what the 'last chance' was.